### DISCUSSION TITLE
Craft 4.4+: Enable tag-like UI for entries field

### DISCUSSION THREAD

**Discussion by tomkiss** [craftcms/cms#12721] [Ideas] [answered]
I've been playing around with the upcoming changes in 4.4. Really enjoying the new categories/globals/tags entrify tools, it works great and makes converting things super easy.

With a content-heavy site, the CP's entries listing contains a load more stuff now, and thankfully this can be tidied up using headings and rearranging the sections neatly.

The only downside to entry-fying content at this point is that I've found the editor experience for tag-like content to be poorer. That is, custom field types that were previously tags are now an entry field and the UX of using the entries field is very different to that of the tags field.

The process for quickly adding lots of associated entries now involves a modal, waiting for content to load within that (this can be slower on larger sites), plus its easiest to use with a mouse.
It's much less slick than the Tags Field, which is good at quick entry using the keyboard only.

My Feature Request is pretty straightforward: Add a "UI Mode" config option to the Entries Field type, that switches the UI to that of the former Tags field. It could be called "Quick search" or something similar.

<img width="406" alt="image" src="https://user-images.githubusercontent.com/183310/220714949-865b451f-a3a8-405b-ac36-7fe24afa8ae8.png">

This would allow the editor experience to basically remain the same as that of tags, which - imo was pretty good.

<img width="345" alt="image" src="https://user-images.githubusercontent.com/183310/220712334-505aedba-89cf-4026-8343-cfe52b9fd41a.png">

**Comment by brandonkelly** [comment]
Sorta similar to #11379. I was hoping to get that in for 4.4 but [hit a snag](https://github.com/craftcms/cms/discussions/11379#discussioncomment-5072729), so may need to wait until 5.0.

**Reply by MoritzLost** [reply]
@brandonkelly Any update on backporting the Tags field UI to entries fields?

I often need something in between Plaintext and Entries fields, where authors should be able to easily enter custom text, but also get autocomplete suggestions from existing values to ensure consistency. The Tags field UI is _perfect_ for this, but now I'm stuck between a rock and a hard place. I don't want to create a Tags field anymore, as they're deprecated. While Tags can be entrified in the future, in terms of UI, the Entries field behaves completely differently from a Tags field. So unless the existing UI/behaviour of Tags fields is added to Entries fields, the Craft 6 upgrade will be a downgrade in terms of AX. But there's also no viable alternative.

Is there anything planned regarding this? Not sure where to go with my current projects that need something like this.

**Reply by brandonkelly** [reply]
I’ve attempted it, but Entries fields’ “Selectable Entries Condition” settings make it pretty complicated. There’s no way to ensure that the created entry is going to match that setting’s condition rules.

Maybe I can revisit, and only add the button for relational fields that don’t define a condition…

**Reply by thupsi** [reply]
IMO that would be perfectly acceptable, since it would be the same with the current tag field, which doesn't support conditions anyway. In other words: The point is that we should be able to replace tags with entries and this solution would fit the bill.

**Reply by thupsi** [reply]
Also, since you say "add the button", we're still talking about the Tags field UI, right? Inline creation of entries is only part of it, but inline search and autocompletion is also key I think.

**Reply by brandonkelly** [reply]
Sorry, I got my wires crossed. The complication with _this_ idea is that we don’t have a way of searching for elements across multiple element sources right now.

So if you have an Entries field with, say, a couple of sections’ sources selected, and maybe even a custom source with an entry condition, we have no way of knowing how to combine the selected sources’ criteria/condition rules together into a single set of query params.

The only way we can make a Tags-esque UI for other relational fields now would be if it’s only available for fields with a single source selected, or if the search input is coupled with a source dropdown you have to select first.

**Reply by thupsi** [reply]
Personaly I'm on board with both ideas (single source/source dropdown). Single source would be on par with current tags field, and source dropdown would be a more advanced version of that. It's a win win!

**Reply by MoritzLost** [reply]
@brandonkelly I'd say only allowing a tags-like UI if a single section would be a good solution. I haven't used tags that much, but every usage could have been replaced with an Entries field targeting a single section. Can't imagine a use-case where I'd need a tags-like UI to target multiple sections. In this case, the regular Entries field UI would be preferable in most cases.

**Reply by mmikkel** [reply]
Seems like limiting the "Tags" UI mode to a single section is the obvious solve, considering the old Tags field only supports a single source (i.e. tag group) as well and this feature is (IMO) primarily about ensuring UI consistency for authors when tags become entrified.

**Comment by wmdhosting** [comment]
![2024-07-08_09-39-56](https://github.com/craftcms/cms/assets/76176291/ec9bc9b3-f143-4443-b880-88c120a3621a)

any chance to add this one on 5.3

**Reply by alexanderbuergin** [reply]
Love to see that one

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.8.0 is out with inline search for relation fields, and inline entry creation for Entries fields (#17497) 🎉

**Reply by tomkiss** [reply]
You beauty! 
Thank-you!
